src/Pkg.jl

```diff
@@ -105,8 +105,8 @@ const PreserveLevel = Types.PreserveLevel
 
 # Define new variables so tab comleting Pkg. works.
 """
-    Pkg.add(pkg::Union{String, Vector{String}}; preserve=PRESERVE_TIERED)
-    Pkg.add(pkg::Union{PackageSpec, Vector{PackageSpec}}; preserve=PRESERVE_TIERED)
+    Pkg.add(pkg::Union{String, Vector{String}}; preserve=PRESERVE_TIERED, installed=false)
+    Pkg.add(pkg::Union{PackageSpec, Vector{PackageSpec}}; preserve=PRESERVE_TIERED, installed=false)
```

Comment:
I don't understand why this extra keyword argument is added. Shouldn't this new feature just be a new `preserve`, `PRESERVE_TIERED_INSTALLED` or something that tells Pkg to first try the installed option and then go for the normal tiered resolve?

Reply:
It's added because the opinion seemed to be that it would be better to not change the default behavior of the tiered resolver. This arg is a specific opt-in, vs. The env var which is general

Reply:
Oh, and map `--installed` to set preserve to `PRESERVE_TIERED_INSTALLED`? That makes sense

Reply:
Updated. I think `pkg> add --preserve=installed Foo` is short enough for local opt-in